crates/primitives/starknet/src/transaction/types.rs

```diff
@@ -220,80 +281,52 @@ impl DeclareTransaction {
 )]
 #[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
 pub struct DeployAccountTransaction {
-    /// Transaction version.
-    pub version: u8,
-    /// Transaction calldata.
-    pub calldata: BoundedVec<Felt252Wrapper, MaxCalldataSize>,
-    /// Account contract nonce.
-    pub nonce: Felt252Wrapper,
-    /// Transaction salt.
-    pub salt: Felt252Wrapper,
-    /// Transaction signature.
-    pub signature: BoundedVec<Felt252Wrapper, MaxArraySize>,
-    /// Account class hash.
-    pub account_class_hash: Felt252Wrapper,
-    /// Max fee.
+    /// The hash identifying the transaction
+    pub transaction_hash: Felt252Wrapper,
```

Comment:
don't add this field in the struct otherwise anyone can set anything he likes when sending the tx using the polkadot sdk